---
title: Comparative Analysis of Feeding Mode and Microbiome Composition in Poison Frog
  Tadpoles
author: "Kayla Weinfurther, Adam Stuckert, Mario Muscarella, Ariane L. Peralta, Kyle Summers"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  pdf_document: null
  word_document: default
  fig_caption: yes
header-includes:
- \usepackage{array}
- \usepackage{graphics}
- \usepackage[utf8]{inputenc}
---

Project Description: Fill out

# Initial Setup
```{r Initial Setup, include=FALSE}
#knitr::opts_chunk$set(echo = TRUE)
#rm(list=ls())
setwd("~/GitHub/TadpoleMicrobiomes/analyses")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code Dependencies
#install.packages('ggplot2')
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
require("vegan")
require("nlme")
require("reshape")
require("ggplot2")
require("ade4")
require("png")
require ("agricolae")
require("tidyr")
require("dplyr")
require("reshape2")
require("picante")
require("tinytex")
```
#Import Files
## Environmental Data
```{r Import Files - Env, include=FALSE}
# Import Environmental Data
design <- read.csv("../data/poison_tadpole_design.csv", row.names=1)
```
## Bacterial Data
```{r Import Files - Bacteria, include=FALSE}
# Import OTU data
# Import Raw Data
otu.in <- read.otu("../data/tadpole.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared")
```

```{r Match design and otu IDs, include=FALSE}
# Correct Sample IDs and Subset File - match up design file and otu files
missing <- setdiff(rownames(design), rownames(otu.in)) 
design_final <- design[-(which(rownames(design) %in% missing)), ]
dim(design_final)
otu <- otu.in
dim(otu)
otu <- otu[match(rownames(design_final), rownames(otu)),]
all.equal(rownames(otu), rownames(design_final))

otus.design <- cbind(design_final,otu)
otus.design.sub <- subset(otus.design, Location != "NA") #removed R. imitator/egg fed/lab
dim(otus.design.sub)

design.final <- otus.design.sub[,c(1:4)] # separate design file after removing NA
otu <- otus.design.sub[,-c(1:4)] # separate otu file after removing NA

# OTU table - remove otus w/ < 2 occurrences across all sites
otu <- otu[, which(colSums(otu) >= 2)]
dim(otu) #went from 8273 to 5761 OTUs

aa <- (rowSums(otu))
aa

tadpole.tax <- read.tax(taxonomy = "../data/tadpole.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy", 
                    format = "rdp", tax.levels = 6, col.tax = 3)
```

# Simple Hypothesis Testing - R. imitator
```{r perMANOVA Bacteria R. imitator, echo=TRUE}
# R. imitator field egg-fed vs. field detritus-fed

otus.design.imitator <- subset(otus.design, Species == "imitator")
dim(otus.design.imitator)

design.final.i <- otus.design.imitator[,c(1:4)] # seaparate design file 
otu.i <- otus.design.imitator[,-c(1:4)] # separate otu file 

# OTU table - remove otus w/ < 2 occurrences across all sites
otu_removal <- otu.i[, which(colSums(otu.i) >= 2)]
dim(otu_removal) # 4696 OTUs 
otu.i <- otu.i[, which(colSums(otu.i) >= 2)]

# Make Relative Abundance Matrices
dataREL.i <- otu.i
for(i in 1:dim(otu.i)[1]){
  dataREL.i[i,] <- otu.i[i,]/sum(otu.i[i,])
}

#PERMANOVA
new.data.i <-cbind(design.final.i,dataREL.i)
adonis = adonis(new.data.i[,-c(1:4)]~Diet, method = "bray", data = new.data.i, perm=1000)
adonis

# Principal Coordinates Analysis
dataREL.dist <- vegdist(dataREL.i, method="bray")

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #20.5
explainvar2b #16.1

pcoa.groups <- paste(new.data.i$Diet, new.data.i$Species, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Diet

#Plot
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=as.factor(pcoa.col), shape=as.factor(pcoa.col))) + theme_bw()
plot1a + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+
  theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") +
  geom_point(aes(colour=as.factor(pcoa.col)), size=5, stroke = 0.75, show.legend = TRUE) + 
    scale_shape_manual(name = "Diet",
                     labels = c("detritus fed","egg fed"),
                     values = c(19, 17))+
  scale_colour_manual(name = "Diet", 
                      labels = c("detritus fed","egg fed"), 
                      values = c("darkgreen","darkorange")) +   
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14),
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (20.5%)") + ylab("PCoA 2 (16.1%)") +
  ggtitle(label="Ranitomeya imitator 16S rRNA Community Composition") + 
  theme(plot.title = element_text(color="black", size=14, face="bold.italic"))
ggsave("../figures/Ordination_imitator_diet.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
ggsave("../figures/Ordination_imitator_diet.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```
# Bacterial community indicator species analysis - R. imitator diet
```{r Bacteria Indicator Species i, echo=TRUE}
library("labdsv")

dataREL <- dataREL.i[, colSums(dataREL.i) > 0.05]
design.type <- new.data.i$Diet
bac.ind <- indval(dataREL, design.type)
levels(design.type)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tadpole.tax[which(as.character(tadpole.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_imitator_diet.txt",
            sep="\t", row.names = F, quote = F)
```

# Simple Hypothesis Testing - R. variabilis
```{r perMANOVA Bacteria R. variabilis, echo=TRUE}
# R. variabilis lab egg-fed vs. field detritus-fed

otus.design.variabilis <- subset(otus.design, Species == "variabilis")
dim(otus.design.variabilis)

design.final.v <- otus.design.variabilis[,c(1:4)] # seaparate design file 
otu.v <- otus.design.variabilis[,-c(1:4)] # separate otu file 

# OTU table - remove otus w/ < 2 occurrences across all sites
otu.v <- otu.v[, which(colSums(otu.v) >= 2)]
dim(otu.v) # 2940 OTUs 

# Make Relative Abundance Matrices
dataREL.v <- otu.v
for(i in 1:dim(otu.v)[1]){
  dataREL.v[i,] <- otu.v[i,]/sum(otu.v[i,])
}

#PERMANOVA
new.data.v <-cbind(design.final.v,dataREL.v)
adonis = adonis(new.data.v[,-c(1:4)]~Diet, method = "bray", data = new.data.v, perm=1000)
adonis

# Principal Coordinates Analysis
dataREL.dist <- vegdist(dataREL.v, method="bray")

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #31.7
explainvar2b #19.3

pcoa.groups <- paste(new.data.v$Diet, new.data.v$Species, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # Diet

#Plot
df2a <- as.data.frame(pcoa.cent.dataframe)
plot1a <- ggplot(df2a, aes(x=V1, y=V2, colour=pcoa.col, shape=pcoa.col)) + theme_bw() 
plot1a + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+
  theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") +
  geom_point(aes(colour=as.factor(pcoa.col)), size=5, stroke = 0.75, show.legend = TRUE) + 
  scale_shape_manual(name = "Diet",
                     labels = c("detritus fed","egg fed"),
                     values = c(19, 17))+
  scale_colour_manual(name = "Diet", 
                      labels = c("detritus fed","egg fed"), 
                      values = c("darkgreen","darkorange")) +  
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14),
          axis.text.x = element_text(size=14), 
          panel.border = element_rect(colour = "black", size=1.25)) + 
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  xlab("PCoA 1 (31.7%)") + ylab("PCoA 2 (19.3%)") +
  ggtitle(label="Ranitomeya variabilis 16S rRNA Community Composition") + 
  theme(plot.title = element_text(color="black", size=14, face="bold.italic"))
ggsave("../figures/Ordination_variabilis_diet.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
ggsave("../figures/Ordination_variabilis_diet.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

# Bacterial community indicator species analysis - R. variabilis diet
```{r Bacteria Indicator Species v, echo=TRUE}
library("labdsv")

dataREL <- dataREL.v[, colSums(dataREL.v) > 0.05]
design.type <- new.data.v$Diet
bac.ind <- indval(dataREL, design.type)
levels(design.type)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tadpole.tax[which(as.character(tadpole.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_variabilis_diet.txt",
            sep="\t", row.names = F, quote = F)
```
# Diversity Metrics - Hypothesis Testing
```{r Diversity Metrics Bacteria calcs, echo=TRUE}
# Rarefy Abundances (min abundance is 13520. We are sampling to 13500)
min(rowSums(otu))
otus.r <- rrarefy(otu, 13500)

# Fisher's Alpha
fisher <- fisher.alpha(otus.r)

# Species Richness
#richness <- rowSums((PWESdata.r >= 1))
richness <- rowSums((otu >= 1))

# Shannon Diversity
shannon <- diversity(otus.r, "shannon")

# Simpson's Evenness
simp.even <- apply(otus.r, 1, simp_even)

#Pielou’s evenness
J <- shannon/log(specnumber(otus.r[,-c(1:1)]))

#combined richness, diversity, evenness
diversity <- cbind(design.final,richness,shannon,simp.even,J)
#write.csv(diversity,"../data/diversity.bact.raw.csv")
```

# Diversity Metrics - Hypothesis Testing - by species
```{r Hypothesis Testing Bacteria both, echo=TRUE}
#summary table for bacterial diversity
summary <- diversity %>% group_by(Species, Diet) %>% summarise(mean.richness=mean(richness), se.richness=se(richness), mean.shannon=mean(shannon), se.shannon=se(shannon))
print(summary)
write.csv(summary,"../data/diversity.bact.summary.csv")

#diversity <- read.csv("../data/diversity.bact.summary.csv", row.names=1)

library(lmerTest) 
# R. imitator
diversity.i <- subset(diversity, Species == "imitator")
richness.lm.i <- lm(richness ~ Diet, data = diversity.i)
richness.lm.i
summary(richness.lm.i)

evenness.lm.i <- lm(simp.even ~ Diet, data = diversity.i)
evenness.lm.i
summary(evenness.lm.i)

shannon.lm.i <- lm(shannon ~ Diet, data = diversity.i)
shannon.lm.i
summary(shannon.lm.i)

# R. variabilis
diversity.v <- subset(diversity, Species == "variabilis")
richness.lm.v <- lm(richness ~ Diet, data = diversity.v)
richness.lm.v
summary(richness.lm.v)

evenness.lm.v <- lm(simp.even ~ Diet, data = diversity.v)
evenness.lm.v
summary(evenness.lm.v)

shannon.lm.v <- lm(shannon ~ Diet, data = diversity.v)
shannon.lm.v
summary(shannon.lm.v)
```

#Plot shannon diversity 
```{r Plot - Shannon Diversity, echo=TRUE}
# Graphing Shannon Diversity
spp.labs <- c("Ranitomeya imitator", "Ranitomeya variabilis") #for facet labels
names(spp.labs) <- c("imitator", "variabilis")

p <- ggplot(diversity, aes(x=Diet, y=shannon, color=as.factor(Diet), shape=as.factor(Diet)))+ geom_boxplot() +
  geom_point(aes(color=factor(Diet)), size=2, position = position_jitterdodge()) +       
  scale_shape_manual(name = "Diet",
                     labels = c("detritus fed","egg fed"),
                     values = c(19, 17))+
  scale_colour_manual(name = "Diet", 
                      labels = c("detritus fed","egg fed"), 
                      values = c("darkgreen","darkorange")) +  
  stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1<-p+facet_wrap(~Species)+facet_grid(. ~ Species,labeller = labeller(Species=spp.labs)) + theme(strip.text = element_text(face = "italic"))
p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Diet", y = "Shannon Diversity Index (H')") + 
    theme(strip.text.x = element_text(size=14, face="italic"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1)) + 
          scale_x_discrete(breaks=c("detritus", "egg fed"), labels=c("detritus fed", "egg fed"))
ggsave("../figures/shannon.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
ggsave("../figures/shannon.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

#Plot Evenness 
```{r Plot Simpson Evenness, echo=TRUE}
# Graphing Simpson's Evenness
p <- ggplot(diversity, aes(x=Diet, y=simp.even, color=as.factor(Diet), shape=as.factor(Diet)))+ geom_boxplot() +
  geom_point(aes(color=factor(Diet)), size=2, position = position_jitterdodge()) +       
  scale_shape_manual(name = "Diet",
                     labels = c("detritus fed","egg fed"),
                     values = c(19, 17)) +
  scale_colour_manual(name = "Diet", 
                      labels = c("detritus fed","egg fed"), 
                      values = c("darkgreen","darkorange")) +  
  stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1<-p+facet_wrap(~Species)+facet_grid(. ~ Species,labeller = labeller(Species=spp.labs)) + theme(strip.text = element_text(face = "italic"))
p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Diet", y = "Simpson's Evenness") + 
    theme(strip.text.x = element_text(size=14, face="italic"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1)) +
          scale_x_discrete(breaks=c("detritus", "egg fed"), labels=c("detritus fed", "egg fed"))
ggsave("../figures/simp.even.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
ggsave("../figures/simp.even.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

#Plot Richness
```{r Plot - OTU richness, echo=TRUE}
# Graphing Richness
p <- ggplot(diversity, aes(x=Diet, y=richness, color=as.factor(Diet), shape=as.factor(Diet)))+ geom_boxplot() +
  geom_point(aes(color=factor(Diet)), size=2, position = position_jitterdodge()) +       
  scale_shape_manual(name = "Diet",
                     labels = c("detritus fed","egg fed"),
                     values = c(19, 17)) +
  scale_colour_manual(name = "Diet", 
                      labels = c("detritus fed","egg fed"), 
                      values = c("darkgreen","darkorange")) +  
  stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+facet_wrap(~Species)+facet_grid(. ~ Species,labeller = labeller(Species=spp.labs))
p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Diet", y = "OTU Richness") + 
    theme(strip.text.x = element_text(size=14, face="italic"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1)) +
          scale_x_discrete(breaks=c("detritus", "egg fed"), labels=c("detritus fed", "egg fed"))
  
ggsave("../figures/richness.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
ggsave("../figures/richness.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

```

```{r Ordination (PCoA) Bacteria all years, eval=FALSE, include=FALSE}
# Bacterial Ordinations - full data set 
# OTU table - remove otus w/ < 2 occurrences across all sites
otu_removal <- otu[, which(colSums(otu) >= 2)]
dim(otu_removal) # 5761 OTUs 
otu <- otu[, which(colSums(otu) >= 2)]
dim(otu)

# Make Relative Abundance Matrices
dataREL <- otu
for(i in 1:dim(otu)[1]){
  dataREL[i,] <- otu[i,]/sum(otu[i,])
}

## Principal Coordinates Ordination
# Principal Coordinates Analysis
dataREL.dist <- vegdist(dataREL, method="bray")

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #19.2
explainvar2b #13.9
```

# Bacterial community composition analysis - Species x Diet
```{r Ordination (PCoA) - Bacteria, eval=FALSE, include=FALSE}
# PERMANOVA
#new.data <-cbind(design.final,dataREL)
new.data <- cbind(design.final,dataREL)
adonis = adonis(new.data[,-c(1:4)]~Diet*Species, method = "bray", data = new.data, perm=1000)
adonis

# Principal Coordinates Analysis
pcoa.groups <- paste(new.data$Diet, new.data$Species, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.spp <- c("imitator","variabilis","imitator","variabilis")
trt.diet <- c("detritus fed", "detritus fed","egg fed","egg fed")
trt.dietspp <- c("detritus_imitator","detritus_variabilis","egg_imitator","egg_variabilis")

pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$diet <- as.factor(trt.diet)
pcoa.cent.dataframe.trts$spp <- as.factor(trt.spp)
pcoa.cent.dataframe.trts$dietspp <- as.factor(trt.dietspp)

#Plot
df1a <- as.data.frame(pcoa.cent.dataframe.trts)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=diet, shape = spp,
                           group = interaction(dietspp))) + theme_bw() 
plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) +
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
geom_point(aes(colour=diet), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("detritus fed","egg fed"), 
                    values = c("darkgreen", "darkorange")) +
scale_fill_manual(labels = c("detritus fed","egg fed"), 
                    values = c("darkgreen", "darkorange")) + 
scale_shape_manual(labels = c("R. imitator","R. variabilis"), 
                   values = c(16,17)) +
theme(axis.title = element_text(size=14), axis.text = element_text(size=14),
      axis.text.x = element_text(size=14), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
xlab("PCoA 1 (22.2%)") + ylab("PCoA 2 (10.1%)") +
  labs(colour = "Diet", shape = "Species") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))
ggsave("../figures/ordination_allspp_diet.pdf", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

# Bacterial community indicator species analysis - Species x Diet
```{r Bacteria Indicator Species all, eval=FALSE, include=FALSE}
library("labdsv")

dataREL <- dataREL[, colSums(dataREL) > 0.05]
design.type <- new.data$Diet
bac.ind <- indval(dataREL, design.type)
levels(design.type)
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

ind.tax <- tadpole.tax[which(as.character(tadpole.tax$OTU) %in% bac.indicators$OTU), ]
ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../data/BacterialIndicators_Diet.txt",
            sep="\t", row.names = F, quote = F)
```